datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -364,6 +367,42 @@ impl LogicalPlan {
     ) -> Result<LogicalPlan, DataFusionError> {
         from_plan(self, &self.expressions(), inputs)
     }
+
+    /// Convert a prepare logical plan into its inner logical plan with all params replaced with their corresponding values
+    pub fn with_param_values(
+        self,
+        param_values: Vec<ScalarValue>,
+    ) -> Result<LogicalPlan, DataFusionError> {
+        match self {
+            LogicalPlan::Prepare(prepare_lp) => {
+                // Verify if the number of params matches the number of values
+                if prepare_lp.data_types.len() != param_values.len() {
+                    return Err(DataFusionError::Internal(format!(
+                        "Expected {} parameters, got {}",
+                        prepare_lp.data_types.len(),
+                        param_values.len()
+                    )));
+                }
+
+                // Verify if the types of the params matches the types of the values
+                for (param_type, value) in
+                    prepare_lp.data_types.iter().zip(param_values.iter())
+                {
+                    if *param_type != value.get_datatype() {
```

Comment:
I think it would help here to add in the index of what parameter had the type mismatch (like if there are 10 parameters and the error says "expected int, got float" it might be harder to debug than "expected int, got float at index 7"

Reply:
Done